### PR TITLE
Atlas adjustments

### PR DESCRIPTION
**Pull request by ubitux**

### CODE REVIEW

File: libnopegl/src/atlas.c [hunk 1]
```diff
@@ -169,10 +169,17 @@ struct texture *ngli_atlas_get_texture(const struct atlas *s)
     return s->texture;
 }
 
-void ngli_atlas_get_dimensions(const struct atlas *s, int32_t *dim)
```
Comment (mbouron) [inline libnopegl/src/atlas.c:172]: Typo `altas` in commit desc.

Comment (mbouron) [reply libnopegl/src/atlas.c:172]: `for an appropriate sampling of the bitmap.`  -> `to sample the bitmap correctly.`

File: libnopegl/src/node_text.c [hunk 2]
```diff
@@ -212,8 +211,8 @@ static int update_character_geometries(struct ngl_node *node)
     }
 
     float *transforms = ngli_calloc(text_nbchr, 4 * 4 * sizeof(*transforms));
-    int32_t *atlas_ids = ngli_calloc(text_nbchr, sizeof(*atlas_ids));
-    if (!transforms || !atlas_ids) {
```
Comment (mbouron) [inline libnopegl/src/node_text.c:216]: You don't need the atlas_id ? even in the future ? (it could be useful for indexing an array from an ubo/ssbo).

Comment (ubitux) [reply libnopegl/src/node_text.c:216]: No plan of using it: all the future parameter will be transmitted individually the same way, through dedicated vertex instance attributes. The benefit over ubo/ssbo is that I don't need to rebuild the whole pipeline when the size changes. If we want to save the pipeline rebuilding with ubo/ssbo, we need to compile a shader with very large arrays for each individual fields. Given how much different fields there will be, it's probably not a good idea to expose a "max size" to the shaders. With instancing, the shader works with individual entries so it's likely more appropriate.

File: libnopegl/src/atlas.c (hunk 1 above)
Comment (ubitux) [reply libnopegl/src/atlas.c:172]: fixed

**Review by mbouron — APPROVED** [review]

**Pull request merged**